### ISSUE TITLE
On right click event causing issues with boats and shields

### ISSUE THREAD

**Issue by Dinoosawruss** [SkriptLang/Skript#2389]
### Description

When using the on right click event no boats, shields of any right click events that a player would normally interact with air work

### Steps to Reproduce

Use on right click event and try to place a boat in the middle of the ocean

### Expected Behavior

Boat places as it should 

### Errors / Screenshots

N/A

### Server Information
* **Server version/platform:**
```[Wed, 21. Aug 2019 19:13:33 EDT INFO] This server is running Paper version git-Paper-174 (MC: 1.14.4) (Implementing API version 1.14.4-R0.1-SNAPSHOT)
[Wed, 21. Aug 2019 19:13:33 EDT INFO] Checking version, please wait...
[Wed, 21. Aug 2019 19:13:34 EDT INFO] Previous version: git-Paper-173 (MC: 1.14.4)
[Wed, 21. Aug 2019 19:13:34 EDT INFO] You are 1 version(s) behind
```
* **Skript version:** 
```
[Wed, 21. Aug 2019 19:14:15 EDT INFO] Skript version 2.4-beta5
[Wed, 21. Aug 2019 19:14:15 EDT INFO] Customize Minecraft's mechanics with simple scripts written in plain English sentences.
[Wed, 21. Aug 2019 19:14:15 EDT INFO] Website: https://skriptlang.github.io/Skript
[Wed, 21. Aug 2019 19:14:15 EDT INFO] Authors: Njol, Mirreski, bensku, TheBentoBox, Snow-Pyon, Pikachu920, Nicofisi, tim740, Tuke-Nuke, TheLimeGlass, xXAndrew28Xx, Sashie, RoyCurtis, nfell2009, Syst3ms and Blueyescat
```
### Additional Context
N/A at this time

**Comment by FranKusmiruk** [comment]
What do you mean by "using" it? If you're cancelling the event, it is supposed to behave like that. Could you provide the trigger you've used to reproduce this issue?

**Comment by masonzettle** [comment]
Hello, I'm the other owner of the server Dino is having issues on. Here's the skript we noticed is causing the issues:
```
on right click:
	if event-item is feather:
		if event-item's name is "&b&l24 Hour Fly Voucher":
			if event-item's lore is "&bGives /fly for 24 hours":
				if player has permission "essentials.fly":
					message "&7&l(&b&l!&f&l) &bYou already have /fly"
					cancel event
					
				else:
					message "&7&l(&b&l!&f&l) &b24 Hour Fly Voucher has been used, use /fly to activate"
					execute console command "/lp user %player% permission settemp essentials.fly true 1d"
					remove feather from the player's tool
					cancel event
					
	if event-item is nametag:
		if event-item's name is "&lOpening Tag":
			if event-item's lore is "&bGives you the &f&lOpening &btag":
				if player has permission "deluxetags.tag.opening":
					message "&7&l(&b&l!&f&l) &bYou already have that tag"
					cancel event
					
				else:
					message "&7&l(&b&l!&f&l) &f&lOpening &btag unlocked, use /tags to activate"
					execute console command "/lp user %player% permission set deluxetags.tag.opening"
					remove nametag from the player's tool
					cancel event
					
	if event-item is sponge:
		if event-item's name contains "Claim Blocks":
			if 1st line of the lore of event-item is "&bGives claim blocks":
				message "&7&l(&b&l!&f&l) &bAdded %2nd line of the lore of event-item% &bClaim Blocks to your total"
				execute console command "/adjustbonusclaimblocks %player% %2nd line of the lore of event-item%"
				remove sponge from the player's tool
				cancel event
```

**Comment by FranKusmiruk** [comment]
Hi, sorry for the late response. From what I can see, there's nothing wrong here, just the fact that you're cancelling the event and thus, cancelling those actions described in the issue.

**Comment by Dinoosawruss** [comment]
The event is only cancelled if the correct conditions are met.

**Comment by Dinoosawruss** [comment]
Great, we've had a lot of people complain to us but at the same time we can't remove the script causing the issue

**Comment by aescraftbr** [comment]
This happens in my server too.
The thing is, I was thinking this was a bug with spigot, and never tried debug.

**Comment by xemnes** [comment]
can confirm this is an issue on my server also, boats cant be place when using the on click event. is there any workaround whilst we wait for a fix?

**Comment by Jordan308** [comment]
I have done some work, and have managed to make a quick fix for it, until it officially gets fixed by the devs... hope this helps with boats atleast

`on right click holding boat:
    if "%targeted block%" contains "water":
        set {_dist} to the distance between location of targeted block and the player
        if {_dist} is less than 4:
            set {_loc} to location of targeted block
            set yaw of {_loc} to yaw of player
            if "%tool of player%" contains "oak":
                spawn oak boat at {_loc}
            if "%tool of player%" contains "spruce":
                spawn oak boat at {_loc}
            if "%tool of player%" contains "birch":
                spawn oak boat at {_loc}
            if "%tool of player%" contains "jungle":
                spawn oak boat at {_loc}        
            if "%tool of player%" contains "acacia":
                spawn oak boat at {_loc}     
            if "%tool of player%" contains "dark oak":
                spawn oak boat at {_loc}`

**Comment by Dinoosawruss** [comment]
@Jordan308 what about shields?

**Comment by Jordan308** [comment]
@Dinoosawruss i've personally not encountered issues with shields.

**Comment by Dinoosawruss** [comment]
@Jordan308 also in that skript is didn’t remove the inventory item but I added it, just a note for anyone using it that you can basically just hold down click and spawn infinite boats

**Comment by xemnes** [comment]
> 
> 
> @Jordan308 also in that skript is didn’t remove the inventory item but I added it, just a note for anyone using it that you can basically just hold down click and spawn infinite boats

try this https://pastebin.com/zCpCa7jA
